stannp/stannp.go

```diff
@@ -141,12 +144,71 @@ func (s *Stannp) post(inputReader io.Reader, inputURL string) (*http.Response, *
 
 	res, err := s.client.Do(req)
 	if err != nil {
-		return nil, util.BuildError(500, fmt.Sprintf("error sending req [%+v]", req), false)
+		return nil, util.BuildError(500, fmt.Sprintf("error sending req [%+v]", req))
 	}
 
 	return res, nil
 }
 
+func (s *Stannp) BytesToPDF(data []byte) (*os.File, *util.APIError) {
+	tmpFile, err := ioutil.TempFile("", "example.*.pdf")
+	if err != nil {
+		return nil, util.BuildError(500, err.Error())
+	}
+
+	if _, writeErr := tmpFile.Write(data); writeErr != nil {
+		closeErr := tmpFile.Close()
+		if closeErr != nil {
+			return nil, util.BuildError(500, closeErr.Error())
+		}
+		return nil, util.BuildError(500, writeErr.Error())
+	}
+
+	return tmpFile, nil
+}
+
+func (s *Stannp) DownloadPDF(pdfURL string) (*letter.PDFRes, *util.APIError) {
+	if !strings.HasPrefix(pdfURL, PDFURLPrefix) {
+		return nil, util.BuildError(400, fmt.Sprintf("pdfURL must begin with [%s]. your input was [%s]", PDFURLPrefix, pdfURL))
```

Comment:
Curious to learn more about the error handling/creation approach in this repo.
It seems like http status codes are being used for non-http related errors. I think this is a product of the fact that most methods return `*util.APIError` rather than a generic `error` interface. If `error` were returned instead (as proposed [here](https://github.com/CopilotIQ/stannp-client-golang/pull/7/files#r1253307620)), we'd be able to change this to
```suggestion
		return nil, fmt.Errorf("pdfURL must begin with [%s]. your input was [%s]", PDFURLPrefix, pdfURL)
```

This pattern could then be applied in a lot of other areas to cleanup the use of `*util.APIError`